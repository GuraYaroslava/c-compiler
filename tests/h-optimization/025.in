int Add(int a, int b)
{
    return a + b;
}

int main()
{
    int a = 1, b = 1;

    int Sub(int a, int b)
    {
        int Mul(int a, int b)
        {
            return a * b;
        }
        return Mul(a, a) - Mul(b, b);
    }

    if (0)
    {
        Sub(1, 3);
    }

    printf("%d", Add(a, 5));
    return 0;
}
