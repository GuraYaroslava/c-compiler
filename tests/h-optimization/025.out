table 0:
`int`
`float`
`char`

table 1:
variable `Add` type of function(variable `a` type of `int`, variable `b` type of `int`) returning `int`
{
return
        a
    + <
        b
}

variable `main` type of function() returning `int`
{
variable `a` type of `int`
variable `b` type of `int`
        a
    = <
        1

        b
    = <
        1

    function:     printf
        format:     "%d"
        args: 
            function:             Add
            args: 
                a
                5



return
    0
}


